File: Objetivo 3/HerancaSimplesExemploPHP/model/ContaEmpresarial.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of ContaEmpresarial
 *
 * @author feijo
 */
class ContaEmpresarial extends Conta{
    private $limite = 30000;
    
    public function getLimite() {
        return $this->limite;
    }

    public function setLimite($limite) {
        $this->limite = $limite;
    }

        
    public function saca($saca) {
        if($saldo >= $saca){
            $saldo-=$saca;
        }else{
            if(($saldo+$Limite)>=$saca){
                $saldo-=$saca;
                $Limite += $saldo;
            }
        }
    }
}

File: Objetivo 3/HerancaSimplesExemploJava/src/model/ContaEmpresarial.java
package model;

/**
 *
 * @author vagner
 */
public class ContaEmpresarial extends Conta{
    
    private double limite = 30000.00;

    public double getLimite() {
        return limite;
    }

    public void setLimite(double limite) {
        this.limite = limite;
    }
    
    @Override
    public void saca(double valor) { //sobreescreve o comportamento da classe Conta
        if((saldo) >= valor){
            saldo -= valor;
        }else if((saldo + limite) >= valor){
            saldo -= valor;
            limite += saldo;
        }
    }
}
